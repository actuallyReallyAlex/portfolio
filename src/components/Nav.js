import React from 'react'
import { Box, Text } from 'grommet'
import { navigationSections } from '../config.json'
import NavLink from './NavLink.js'

const Nav = () => {
  return (
    <Box
      align="center"
      className="primary-color"
      direction="row"
      fill="horizontal"
      height="50px"
      justify="between"
      pad={{ horizontal: 'medium' }}
    >
      <Text color="white" size="xlarge" weight="bold">
        Alex Lee
      </Text>
      <Box direction="row" gap="small">
        {navigationSections.map(title => (
          <NavLink key={title} title={title} />
        ))}
      </Box>
    </Box>
  )
}

export default Nav
