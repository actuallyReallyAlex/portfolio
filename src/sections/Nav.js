import React from 'react'

const Section = () => {
  return <h1>Section</h1>
}

export default Section
