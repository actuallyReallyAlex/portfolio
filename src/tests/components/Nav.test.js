import React from 'react'
import Nav from '../../components/Nav'
import { shallow } from 'enzyme'

const defaultProps = {
  navigationSections: ['Home', 'About', 'Résumé', 'Portfolio'],
  size: 'medium',
  views: [
    { title: 'Home', isInView: true },
    { title: 'About', isInView: false },
    { title: 'Résumé', isInView: false },
    { title: 'Portfolio', isInView: false }
  ]
}

describe('<Nav />', () => {
  test('Should render the <Nav /> component.', () => {
    const component = shallow(<Nav {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  test('Should render the <Nav /> component when viewed from a small device.', () => {
    const component = shallow(<Nav {...defaultProps} size="small" />)
    expect(component).toMatchSnapshot()
  })
})
