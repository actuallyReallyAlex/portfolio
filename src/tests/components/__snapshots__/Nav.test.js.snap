// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> Should render the <Nav /> component when viewed from a small device. 1`] = `ShallowWrapper {}`;

exports[`<Nav /> Should render the <Nav /> component. 1`] = `ShallowWrapper {}`;
